packages/@aws-cdk/aws-apigateway/lib/method.ts

```diff
@@ -117,14 +119,26 @@ export class Method extends Resource {
 
     const defaultMethodOptions = props.resource.defaultMethodOptions || {};
     const authorizer = options.authorizer || defaultMethodOptions.authorizer;
+    const authorizationType = options.authorizationType || defaultMethodOptions.authorizationType || AuthorizationType.NONE;
+
+    if (authorizer && authorizationType !== AuthorizationType.CUSTOM) {
+      throw new Error(`${this.resource}/${this.httpMethod} - Authorization type is not set to 'CUSTOM' when an authorizer is specified`);
+    }
+    if (!authorizer && authorizationType === AuthorizationType.CUSTOM) {
+      throw new Error(`${this.resource}/${this.httpMethod} - Authorizer is not specified when the authorization type is set to 'CUSTOM'`);
+    }
```

Comment:
These validations tell me that we should probably deprecate `AuthorizationType` altogther and have `IAuthorizer` report it's type.